Controllers/CustomersController.cs

```diff
@@ -0,0 +1,208 @@
+        public async Task<IActionResult> ListCustomers(int? id)
+        {
+            var Customers = await _context.Customers
+                .Include(b => b.Entity)
+                .Include(b => b.Contact)
+                .Where(m => m.EntityId == id)
+                .ToListAsync();
+
+            var customersVM = new List<CustomerFormViewModel>();
+
+            foreach (var customer in Customers)
+            {
+                customersVM.Add(_mapper.Map<CustomerFormViewModel>(customer));
+            }
+
+            return View(customersVM);
+        }
+
+        // GET: Customers/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var customer = await _context.Customers
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            if (customer == null)
+            {
+                return NotFound();
+            }
+
+            return View(_mapper.Map<CustomerFormViewModel>(customer));
+        }
+
+        // GET: Customers/Create
+        public IActionResult Create()
+        {
+            var customerVM = new CustomerFormViewModel
+            {
+                Entities = _context.Entities.ToList()
+            };
+
+            return View(customerVM);
+        }
+
+        // POST: Customers/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create(CustomerFormViewModel customerVM)
+        {
+            if (ModelState.IsValid)
+            {
+                var customer = _mapper.Map<Customer>(customerVM);
+
+                _context.Add(customer);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            customerVM.Entities = _context.Entities.ToList();
+
+            return View(customerVM);
+        }
+
+        // GET: Customers/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var customer = await _context.Customers
+                .Include(b => b.Contact)
+                .Include(b => b.Entity)
+                .FirstOrDefaultAsync(m => m.Id == id);
+
+            if (customer == null)
+            {
+                return NotFound();
+            }
+
+            var customerVM = _mapper.Map<CustomerFormViewModel>(customer);
+
+            customerVM.Entities = _context.Entities.ToList();
+
+            return View(customerVM);
+        }
+
+        // POST: Customers/Edit/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, CustomerFormViewModel customerVM)
+        {
+            if (id != customerVM.Id)
+            {
+                return NotFound();
```

Comment:
In my opinion it is not HTTP 404 NotFound that stands for unavailiable resource, but rather HTTP 400 BadRequest.
Please look brifely:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

To go deeper status codes are defined by section 10 of RFC 2616. You can find an updated specification in RFC 7231.